### ISSUE TITLE
No output on USB Serial JTAG with IDF 5.2.2 (IDFGH-13085)

### ISSUE THREAD

**Issue by pggh** [espressif/esp-idf#14031]
### Answers checklist.

- [X] I have read the documentation [ESP-IDF Programming Guide](https://docs.espressif.com/projects/esp-idf/en/latest/) and the issue is not addressed there.
- [X] I have updated my IDF branch (master or release) to the latest version and checked that the issue is present there.
- [X] I have searched the issue tracker for a similar issue and not found a similar issue.

### IDF version.

5.2.2

### Espressif SoC revision.

ESP32-S3

### Operating System used.

Windows

### How did you build your project?

Command line with Make

### If you are using Windows, please specify command line type.

None

### Development Kit.

own

### Power Supply used.

USB

### What is the expected behavior?

Getting the output data.

### What is the actual behavior?

Starting with IDF 5.2.2, I get almost no output on the USB Serial JTAG, when I enter my command-line repl loop.
only sometimes a few characters will can be seen on the output. But all log-output as well as the command-prompt and input echo is basically gone. Same code with IDF 5.2.1 works.

### Steps to reproduce.

As part of my own command-line repl loop implementation, I do perform the following steps:

```
   /* Disable buffering on stdin */
   setvbuf(stdin, NULL, _IONBF, 0);

   /* Minicom, screen, idf_monitor send CR when ENTER key is pressed */
   esp_vfs_dev_usb_serial_jtag_set_rx_line_endings(ESP_LINE_ENDINGS_CR);
   /* Move the caret to the beginning of the next line on '\n' */
   esp_vfs_dev_usb_serial_jtag_set_tx_line_endings(ESP_LINE_ENDINGS_CRLF);

   usb_serial_jtag_driver_config_t usb_serial_jtag_config = USB_SERIAL_JTAG_DRIVER_CONFIG_DEFAULT();

   /* Install USB-SERIAL-JTAG driver for interrupt-driven reads and writes */
   ESP_ERROR_CHECK(usb_serial_jtag_driver_install(&usb_serial_jtag_config));

   // flush input fifo, otherwise driver never gets an interrupt for new data because fifo already full
   uint8_t c;
   while (usb_serial_jtag_ll_read_rxfifo(&c, 1) != 0) {}

   esp_vfs_usb_serial_jtag_use_nonblocking();
```
Note: the use of usb_serial_jtag_driver_install and esp_vfs_usb_serial_jtag_use_nonblocking is required to have
a task-blocking usb_serial_jtag_read_bytes and a non-buffered output (no buffer, busy-waiting, not task-blocking)

### More Information.

I don't know if this is related to [this](https://github.com/espressif/esp-idf/issues/13939) issue.
The symptoms in my case is different; almost no output vs. only some chars missing.

Input is still processed but command and log-output is missing. As I don't have very long (log-) lines, it could be that statement

> I also noticed that adding more fsync(fileno(stdout)) would make the situation worse

could match if 'worse' means discard all buffered data.

Basically 5.2.2 is unusable because of this bug. (5.2.1 is unusable because of other bugs line I2C)

**Comment by mythbuster5** [comment]
it has been fixed by commit b19f9242
